src/components/timecard/time-input-errors/TimeInputErrors.jsx

```diff
@@ -0,0 +1,118 @@
+import { PropTypes } from 'prop-types';
+import { useApplicationContext } from '../../../context/ApplicationContext';
+import {
+  formatLongDate,
+  formatTime,
+  getTimePeriodTypesMap,
+} from '../../../utils/time-entry-utils/timeEntryUtils';
+
+const TimeInputErrors = ({ clashingProperty, clashes }) => {
+  const { timePeriodTypes } = useApplicationContext();
+  const timePeriodTypesMap = getTimePeriodTypesMap(timePeriodTypes);
+
+  const displayClashingProperty = () => {
+    if (clashingProperty === 'startTime') {
+      return <p>Your start time must not overlap with another time period</p>;
+    }
+
+    if (clashingProperty === 'endTime') {
+      return <p>Your finish time must not overlap with another time period</p>;
+    }
+
+    return (
+      <p>
+        Your start and finish times must not overlap with another time period
+      </p>
+    );
+  };
+
+  const displaySingleTimeClash = () => {
+    const clash = clashes[0];
+    let text;
+
+    const timePeriodType = timePeriodTypesMap[clash.timePeriodTypeId];
+
+    if (timePeriodType === 'Shift') {
+      text = `You are already assigned to work from ${startDateTime(
+        clash
+      )} to ${formatTime(clash.endTime)} on ${formatLongDate(clash.endTime)}`;
+    } else {
+      text = `You are already assigned a ${timePeriodType.toLowerCase()} on ${formatLongDate(
+        clash.startTime
+      )}`;
+    }
+
+    return <p>{text}</p>;
+  };
+
+  const displayMultipleTimeClashes = () => {
+    sortClashes(clashes);
+
+    const times = (
+      <div>
+        <p>You are already assigned to the following time periods:</p>
+        <ul>
+          {clashes.map((clash, index) => (
+            <li key={index} data-testid="test">
+              {timePeriodClashToText(clash)}
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+    return times;
+  };
+
+  const sortClashes = (clashes) => {
+    return clashes.sort(
+      (a, b) => Date.parse(a.startTime) - Date.parse(b.startTime)
+    );
+  };
+
+  const timePeriodClashToText = (clash) => {
+    const timePeriodType = timePeriodTypesMap[clash.timePeriodTypeId];
+
+    if (timePeriodType === 'Shift') {
+      return `${startDateTime(clash)} to ${formatTime(
+        clash.endTime
+      )} on ${formatLongDate(clash.endTime)}`;
+    }
+
+    return `${timePeriodType.toLowerCase()} on ${formatLongDate(
+      clash.startTime
+    )}`;
+  };
+
+  const startDateTime = (clash) => {
+    const startTime = new Date(clash.startTime);
+    const endTime = new Date(clash.endTime);
+    let startDate = '';
+
+    if (startTime.toDateString() !== endTime.toDateString()) {
```

Comment:
any specific reason for converting the date to a textual string and then comparing?
Can we use `getTime()` instead?

Reply:
This is checking if they are on the same day, rather than if they are the same day and time. We decided against using `getDate` as it only returns the day of the month, not an actual date. We could do something like this (https://stackoverflow.com/a/43855248), but it seemed quite long winded?

Reply:
Right I see, can we possibly use dayJs instead and use the `isSame` function and pass in `day` as the second param. 
It looks like that would internally compare day, month, and year (like SO ex) but in a concise line for us. Plus it would keep consistency & simplicity with using dayJs library.
https://day.js.org/docs/en/query/is-same

If that somehow ends up consequently over complicating other parts then I think it's fine as it is. But yeah might be worth a try.

Reply:
Updated to use dayjs :)

Reply:
Nice, minor further suggestion is to remove the javascript date objects too (ln 94,95). 
I think you should be able to just pass in the times from the clash object to dayjs directly.
E.g
`if (!dayjs(clash.startTime).isSame(dayjs(clash.endTime), 'day')) {`

Again feel free to just resolve if short for time / doesn't work.

Reply:
👍 done